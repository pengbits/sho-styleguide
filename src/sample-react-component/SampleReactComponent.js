import React, { Component } from 'react';
import classNames from 'classnames';

class SampleReactComponent extends Component {
	constructor(props) {
		super(props);
	}
	
	_onClick() {
		const { dispatchUpdateLike } = this.props;
		dispatchUpdateLike()
	}
	
	render() {
		return (
			<div>
				<h2 className="sample-react-component__header">This is a Sample React Component</h2>
				<div className="sample-react-component__button" onClick={this._onClick.bind(this)}>
					<i className={classNames("sample-react-component__button-icon",
											 						 { "sample-react-component__button-icon--thumbsup": this.props.like,
											 	 						 "sample-react-component__button-icon--thumbsdown": !this.props.like
											 	 					 })}>
					</i>
				</div>
			</div>	
		)
	}
}

export default SampleReactComponent