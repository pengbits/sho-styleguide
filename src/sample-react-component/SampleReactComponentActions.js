import { createAction } from 'redux-actions';

export const UPDATE_LIKE = 'UPDATE_LIKE';

export const updateLike = createAction(UPDATE_LIKE)