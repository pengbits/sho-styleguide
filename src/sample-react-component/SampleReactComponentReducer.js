import { UPDATE_LIKE } from './SampleReactComponentActions';

const SampleReactComponentReducer = (state = { like: true }, action) => {
	switch (action.type) {
		case UPDATE_LIKE:
			return {
				like: !state.like
			}
		default:
			return state
	}
}

export default SampleReactComponentReducer