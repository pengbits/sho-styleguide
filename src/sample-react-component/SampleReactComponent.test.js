import React from 'react';
import { shallow, mount } from 'enzyme';
import SampleReactComponent from './SampleReactComponent';
import { UPDATE_LIKE, updateLike } from './SampleReactComponentActions';
import { default as reducer } from './SampleReactComponentReducer';

describe('<SampleReactComponent />', () => {
	test('renders', () => {
		let wrapper = shallow(<SampleReactComponent />)
		expect(wrapper).toHaveLength(1);
	})

	test('matches snapshot', () => {
		let wrapper = mount(<SampleReactComponent />)
		expect(wrapper).toMatchSnapshot();
	})

	describe('Redux', () => {
		test('updateLike() returns a type and payload', () => {
			let actual = updateLike(true);
			let expected = { 	
				type: UPDATE_LIKE,
				payload: true
		  };
			expect(actual).toEqual(expected);
		})

		test('reducer returns state with flipped boolean', () => {
			let initialState = { like: true };
			let action = updateLike(true);
			let actual = reducer(initialState, action);
			let expected = { like: false };
			expect(actual).toEqual(expected);
		})
	})
})