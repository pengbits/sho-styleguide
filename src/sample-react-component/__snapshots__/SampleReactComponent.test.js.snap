// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SampleReactComponent /> matches snapshot 1`] = `
<SampleReactComponent>
  <div>
    <h2
      className="sample-react-component__header"
    >
      This is a Sample React Component
    </h2>
    <div
      className="sample-react-component__button"
      onClick={[Function]}
    >
      <i
        className="sample-react-component__button-icon sample-react-component__button-icon--thumbsdown"
      />
    </div>
  </div>
</SampleReactComponent>
`;
